File: fix_literals.py
import re
import sys

def transpile_template_literals(content):
    # Find all template literals
    def replace_literal(match):
        literal = match.group(0)
        # Remove the backticks
        inner = literal[1:-1]
        
        # Split by ${...}
        parts = []
        last_end = 0
        
        # Regex to find ${...}
        for m in re.finditer(r'\$\{([^}]+)\}', inner):
            # Text before the variable
            text = inner[last_end:m.start()]
            if text:
                # Escape quotes and newlines for normal string
                text = text.replace('\n', '\\n').replace('"', '\\"')
                parts.append(f'"{text}"')
            
            # The variable
            var = m.group(1)
            parts.append(f"({var})")
            
            last_end = m.end()
            
        # Text after the last variable
        text = inner[last_end:]
        if text:
            text = text.replace('\n', '\\n').replace('"', '\\"')
            parts.append(f'"{text}"')
            
        if not parts:
            return '""'
            
        return ' + '.join(parts)

    return re.sub(r'`[^`]*`', replace_literal, content)

with open('cart.js', 'r', encoding='utf-8') as f:
    content = f.read()

new_content = transpile_template_literals(content)

with open('cart.js', 'w', encoding='utf-8') as f:
    f.write(new_content)
print("Done")

File: fix_literals.js
const fs = require('fs');

let content = fs.readFileSync('cart.js', 'utf-8');

// Replace template literals with string concatenations
content = content.replace(/`([\s\S]*?)`/g, function(match, inner) {
    let parts = [];
    let regex = /\$\{([^}]+)\}/g;
    let lastIndex = 0;
    let m;
    
    while ((m = regex.exec(inner)) !== null) {
        let text = inner.substring(lastIndex, m.index);
        if (text) {
            // Escape double quotes and newlines
            text = text.replace(/\\/g, '\\\\').replace(/"/g, '\\"').replace(/\r?\n/g, '\\n" +\n"');
            parts.push('"' + text + '"');
        }
        parts.push('String(' + m[1] + ')');
        lastIndex = regex.lastIndex;
    }
    
    let text = inner.substring(lastIndex);
    if (text) {
        text = text.replace(/\\/g, '\\\\').replace(/"/g, '\\"').replace(/\r?\n/g, '\\n" +\n"');
        parts.push('"' + text + '"');
    }
    
    if (parts.length === 0) return '""';
    return parts.join(' + ');
});

fs.writeFileSync('cart.js', content, 'utf-8');
console.log('Template literals replaced!');
